### ISSUE TITLE
Build failure using Swift 4.2 from Xcode 10 Beta

### ISSUE THREAD

**Issue by wlisac** [SwiftGen/StencilSwiftKit#92]
I'm running into an issue building StencilSwiftKit with SPM using Swift 4.2.

```
/.build/checkouts/StencilSwiftKit.git-5443537928707802040/Sources/SwiftIdentifier.swift:16:55: error: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions
private let headRanges: [CountableClosedRange<Int>] = [
```

It looks like the same issue was detected by the Swift source compatibility suite: https://bugs.swift.org/browse/SR-8046

Has anyone else seen this / know of a work around?

**Comment by TimeDelta** [comment]
I ran into this as well when trying to build Sourcery: https://github.com/krzysztofzablocki/Sourcery/issues/649

**Comment by AliSoftware** [comment]
I understand the error and can see that this array of CountableRanges is big, but I'm not sure how to make that easier to parse for the Swift compiler and type checker…

Any idea how to annotate that better to help the type checker and fix this?

**Comment by AliSoftware** [comment]
I've created a PR to fix this.

I've tested it locally with Xcode 10b3 (I haven't had time downloading b4 yet) and this seems to fix it in the meantime, by adding hints all over the place for the type checker.

Unfortunately, we haven't configured CircleCI to test under Xcode 10 just yet (currently it still tests under Xcode 9.2). Might be interesting to do a PR so that Circle checks both on Xcode 9.4 and 10 in the future
